test/gtest/cartesianprod.hpp

```diff
@@ -0,0 +1,300 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "cpu_cartesianprod.hpp"
+#include "get_handle.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include <cstdio>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <miopen/cartesianprod.hpp>
+#include <miopen/miopen.h>
+
+template <class T>
+inline std::ostream& operator<<(std::ostream& os, const std::vector<T>& v)
+{
+    os << '{';
+    for(int i = 0; i < v.size(); ++i)
+    {
+        if(i != 0)
+            os << ',';
+        os << v[i];
+    }
+    os << '}';
+    return os;
+}
+
+struct CartesianProdTestCase
+{
+    std::vector<std::vector<size_t>> inputs_dims;
+    bool is_contiguous = true;
+    friend std::ostream& operator<<(std::ostream& os, const CartesianProdTestCase& tc)
+    {
+        os << " input_dims:";
+        for(int i = 0; i < tc.inputs_dims.size(); i++)
+        {
+            auto input = tc.inputs_dims[i];
+            if(i != 0)
+                os << ",";
+            os << input[0];
+            for(int j = 1; j < input.size(); j++)
+            {
+                os << "x" << input[j];
+            }
+        }
+        return os << " is_contiguous:" << tc.is_contiguous;
+    }
+
+    const std::vector<std::vector<size_t>>& GetInputs() const { return inputs_dims; }
+    std::vector<size_t> ComputeStrides(std::vector<size_t> inputDim) const
+    {
+        if(!is_contiguous)
+            std::swap(inputDim.front(), inputDim.back());
+        std::vector<size_t> strides(inputDim.size());
+        strides.back() = 1;
+        for(int i = inputDim.size() - 2; i >= 0; --i)
+            strides[i] = strides[i + 1] * inputDim[i + 1];
+        if(!is_contiguous)
+            std::swap(strides.front(), strides.back());
+        return strides;
+    }
+};
+
+inline std::vector<CartesianProdTestCase> CartesianProdTestConfigs()
+{
+    return {
+        {{{5}, {7}, {9}}, true},
+        {{{6}, {4}, {9}}, false},
+    };
+}
```

Comment:
As a single `input` tensor is 1D, `inputs_dims` should be `std::vector<size_t>`

Reply:
resolved